webapp/src/modules/parcels/utils.js

```diff
@@ -1,6 +1,21 @@
-export function toParcelObject(parcelsArray) {
+const emptyParcel = {
+  x: null,
+  y: null,
+  price: null,
+  owner: null,
+  district_id: null,
+  data: {
+    name: '',
+    description: '',
+    ipns: null,
+    version: 0
+  }
+}
+
+export function toParcelObject(parcelsArray, allParcels) {
   return parcelsArray.reduce((map, parcel) => {
-    map[parcel.id] = parcel
+    const oldParcel = (allParcels && allParcels[parcel.id]) || emptyParcel
+    map[parcel.id] = { ...oldParcel, ...parcel }
```

Comment:
What do you think of simplifying this a bit:

```javascript
export function toParcelObject(parcelsArray, allParcels = {}) {
  return parcelsArray.reduce((map, parcel) => {
    const oldParcel = allParcels[parcel.id]
    map[parcel.id] = { ...emptyParcel, ...oldParcel, ...parcel }
    return map
  }, {})
}
```

I **don't** want to impose the implementation, just as a discussion